frontend/src/ui/components/PublishButton/PublishButton.tsx

```diff
@@ -0,0 +1,50 @@
+import * as React from 'react';
+import { WrappedMessage } from 'utils/intl';
+import { Button } from 'react-bootstrap';
+import messages from './displayMessages';
+import './styles.scss';
+
+interface Props {
+  deploymentDisabled: boolean;
+  undeployedChanges: number;
+  onClickWrapper: Function;
+}
+
+export const PublishButton: React.FC<Props> = ({
+  deploymentDisabled,
+  undeployedChanges,
+  onClickWrapper
+}: Props) => {
+  const notificationNumber = undeployedChanges <= 9 ? undeployedChanges : '9+';
+
+  return (
+    <Button
+      className={
+        deploymentDisabled
+          ? 'float-right disabledBtn'
+          : 'float-right enabledBtn'
+      }
+      variant="primary"
+      size="lg"
+      disabled={deploymentDisabled}
+      onClick={() => {
+        onClickWrapper();
+      }}
+    >
+      {undeployedChanges !== 0 && (
+        <div className="notificationLayer">
+          <p>
+            <WrappedMessage
+              id="notificationText"
+              messages={messages}
+              values={{ notificationNumber }}
+            />
+          </p>
+        </div>
+      )}
+      <div className="textLayer">
```

Comment:
can we name the class better and meaningful?

Reply:
Changed to buttonTextLayer